google/cloud/aiplatform/jobs.py

```diff
@@ -724,7 +724,8 @@ def iter_outputs(
 
             # Build a Storage Client using the same credentials as JobServiceClient
             storage_client = storage.Client(
-                credentials=self.api_client._transport._credentials
+                project=self.project,
```

Comment:
Should the project be the same as the project for the resource as the config's project may be set to a different project? Is it a safe assumption that the jobs outputs will be written to the same project as the resource?